packages/go/cypher/analyzer/analyzer.go

```diff
@@ -125,25 +152,31 @@ func (s *ComplexityMeasure) onProjection(node *model.Projection) {
 		// Distinct incurs a weight since it will change how the projection is materialized
 		s.Weight += Weight1
 	}
+
+	return nil
 }
 
-func (s *ComplexityMeasure) onPartialComparison(node *model.PartialComparison) {
+func (s *ComplexityMeasure) onPartialComparison(stack *model.WalkStack, node *model.PartialComparison) error {
 	switch node.Operator {
 	case model.OperatorRegexMatch:
 		// Regular expression matching incurs a weight since it can be far more involved than any of the other
 		// string operators
 		s.Weight += Weight1
 	}
+
+	return nil
 }
 
-func (s *ComplexityMeasure) onNodePattern(node *model.NodePattern) {
-	if node.Binding == "" {
+func (s *ComplexityMeasure) onNodePattern(stack *model.WalkStack, node *model.NodePattern) error {
```

Comment:
I can't remember if it's a compiler warning or not but you may want to consider renaming "stack" to "_" since that param is unused.

Reply:
Patched up